addons/stock_account/models/product.py

```diff
@@ -317,7 +317,7 @@ def _change_standard_price(self, new_price):
             }
             am_vals_list.append(move_vals)
 
-        account_moves = self.env['account.move'].sudo().with_context(clean_context(self._context)).create(am_vals_list)
+        account_moves = self.with_context(clean_context(self._context)).env['account.move'].sudo().with_context(force_zero_quantity=True).create(am_vals_list)
```

Comment:
You can do both those things in a single `with_context()`
```suggestion
        account_moves = self.with_context(clean_context(self._context), force_zero_quantity=True).env['account.move'].sudo().create(am_vals_list)
```